cli/src/lib/plugin/webpack.config.js

```diff
@@ -0,0 +1,316 @@
+                Object.assign(
+                    {
+                        inject: true,
+                        filename: paths.pluginLaunchPath,
+                        template: paths.shellPublicPluginHtml,
+                    },
+                    isProduction
+                        ? {
+                              minify: {
+                                  removeComments: true,
+                                  collapseWhitespace: true,
+                                  removeRedundantAttributes: true,
+                                  useShortDoctype: true,
+                                  removeEmptyAttributes: true,
+                                  removeStyleLinkTypeAttributes: true,
+                                  keepClosingSlash: true,
+                                  minifyJS: true,
+                                  minifyCSS: true,
+                                  minifyURLs: true,
+                              },
+                          }
+                        : undefined
+                )
+            ),
+            isProduction &&
+                new MiniCssExtractPlugin({
+                    filename: 'static/css/[name].[contenthash:8].css',
+                    chunkFilename:
+                        'static/css/[name].[contenthash:8].chunk.css',
+                }),
+            // Makes some environment variables available to the JS code, for example:
+            // if (process.env.NODE_ENV === 'production') { ... }.
+            // It is absolutely essential that NODE_ENV is set to production
+            // during a production build.
+            // Otherwise React will be compiled in the very slow development mode.
+            new webpack.DefinePlugin({
+                'process.env': {
+                    ...Object.keys(shellEnv).reduce((env, key) => {
+                        env[key] = JSON.stringify(shellEnv[key])
+                        return env
+                    }, {}),
+                    NODE_ENV: JSON.stringify(webpackEnv),
+                },
+            }),
+            new webpack.IgnorePlugin({
+                resourceRegExp: /^\.\/locale$/,
+                contextRegExp: /moment$/,
+            }),
+            // dhis2: Inject plugin static assets to the existing SW's precache manifest
+            process.env.NODE_ENV === 'production' &&
+                new WorkboxWebpackPlugin.InjectManifest({
+                    swSrc: paths.shellBuildServiceWorker,
+                    injectionPoint: 'self.__WB_PLUGIN_MANIFEST',
+                    // Skip compiling the SW, which happens in the app build step
+                    compileSrc: false,
+                    dontCacheBustURLsMatching: /\.[0-9a-f]{8}\./,
+                    exclude: [
+                        /\.map$/,
+                        /asset-manifest\.json$/,
+                        /LICENSE/,
+                        // TODO dhis2: locales are weird in the plugin build -
```

Comment:
Any idea why?

Reply:
I think I wrote this because all the generated locales files are duplicated (but slightly different); one batch in a `moment-locales` dir and another in `plugin-moment-locales`

I think we talked about not precaching locales files too, conveniently

It's been a while since I looked at this, so I forget if there was anything else 🤔